wagtail/contrib/modeladmin/helpers/permission.py

```diff
@@ -28,6 +31,10 @@ def get_all_model_permissions(self):
             content_type__model=self.opts.model_name,
         )
 
+    @cached_property
+    def all_permission_codenames(self):
+        return self.get_all_model_permissions().values_list("codename", flat=True)
```

Comment:
Nit: Save caching a large queryset object, we can just get the list of strings:

```suggestion
        return list(self.get_all_model_permissions().values_list("codename", flat=True))
```
And, perhaps making it distinct:

```suggestion
        return list(self.get_all_model_permissions().values_list("codename", flat=True).distinct())
```